server/manifestmergetool/src/main/java/com/defold/manifestmergetool/ManifestMergeTool.java

```diff
@@ -59,44 +32,26 @@ public void usage() {
     }
 
     private static void mergeAndroid(File main, File[] libraries, File out) throws RuntimeException {
-        // Good explanation of merge rules: https://android.googlesource.com/platform/tools/base/+/jb-mr2-dev/manifest-merger/src/main/java/com/android/manifmerger/ManifestMerger.java
-        // https://github.com/pixnet/android-platform-tools-base/blob/master/build-system/manifest-merger/src/main/java/com/android/manifmerger/ManifestMerger2.java
-        // https://android.googlesource.com/platform/tools/base/+/master/build-system/manifest-merger/src/main/java/com/android/manifmerger/Merger.java
-        logger.log(Level.FINE, "Merging manifests for Android");
-
-        ManifestMerger2.Invoker invoker = ManifestMerger2.newMerger(main, ManifestMergeTool.androidLogger, ManifestMerger2.MergeType.APPLICATION);
-        invoker.addLibraryManifests(libraries);
-        invoker.withFeatures(ManifestMerger2.Invoker.Feature.REMOVE_TOOLS_DECLARATIONS);
-
-        try {
-            MergingReport report = invoker.merge();
-            if (report.getResult().isSuccess()) {
-                String xml = report.getMergedDocument(MergingReport.MergedManifestKind.MERGED);
-                if (out != null) {
-                    try {
-                        try (OutputStreamWriter writer = new OutputStreamWriter(new FileOutputStream(out), StandardCharsets.UTF_8)) {
-                            writer.write(xml);
-                        }
-                    } catch (IOException e) {
-                        throw new RuntimeException("Failed to write xml to " + out.getAbsolutePath(), e);
-                    }
-                } else {
-                    System.out.println(xml);
-                }
-            } else {
-                report.log(ManifestMergeTool.androidLogger);
-                throw new RuntimeException("Failed to merge manifests: " + report.toString());
-            }
-        } catch (ManifestMerger2.MergeFailureException e) {
-            throw new RuntimeException("Exception while merging manifests: " + e.toString());
-        }
+        AndroidManifestMerger merger = new AndroidManifestMerger(logger);
+        merger.merge(main, libraries, out);
     }
 
     private static void mergePlist(File main, File[] libraries, File out) throws RuntimeException {
         InfoPlistMerger merger = new InfoPlistMerger(logger);
         merger.merge(main, libraries, out);
     }
 
+    private static void mergeApple(File main, File[] libraries, File out) throws RuntimeException {
```

Comment:
Send .xcprivacy files to a bespoke merger. Use InfoPlistMerger for everything else.